src/lib/selector/Query.ts

```diff
@@ -29,6 +29,20 @@ export class Query {
         }
     }
 
+    public static async executeQueries(connection: Connection, queries: string[], logLabel: string, recordsCount?: number) {
+        const recordResults = [];
+        for (const query of queries) {
+            try {
+                const result = await this.executeQuery(connection, query, logLabel, recordsCount);
+                recordResults.push(result);
+            } catch (error) {
+                Logs.displayError(Logs.addEnxooMessages(error));
+                throw new Error('Error during query execution. Importer will now exit the operation.');
+            }
+        }
+        return recordResults.flat();
+    }
+
     public static async executeBulkQuery(connection: Connection, query: string, logLabel: string) {
```

Comment:
Wanted to use bulk v2 for bulk query, however, there is an open error in the plugin for a large number of records. After the ticket fix it is possible to change to v2
https://github.com/jsforce/jsforce/issues/1546

![image](https://github.com/user-attachments/assets/0faa7bcc-84ce-496a-95b2-6acefa0daa6f)